### ISSUE TITLE
arm64 all{mod,yes}config failure on next-20190830

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#648]
```
aarch64-linux-gnu-ld: mm/memcontrol.o: in function `__cmpxchg_mb':
memcontrol.c:(.text+0x1898): undefined reference to `__compiletime_assert_175'
```

This appears to come from https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/arch/arm64/include/asm/cmpxchg.h?id=529952053dc1b27702fc762abe00f30fe11ce715#n166.

Bisect proved inconclusive (it appears to be an interaction between two trees because it points to a merge commit: https://git.kernel.org/next/linux-next/c/529952053dc1b27702fc762abe00f30fe11ce715).

```
# bad: [6d028043b55e54f48fbdf62ea8ce11a4ad830cac] Add linux-next specific files for 20190830
# good: [4a64489cf8e21a17fd6dd88935818ba7307ba996] Merge tag 'Wimplicit-fallthrough-5.3-rc7' of git://git.kernel.org/pub/scm/linux/kernel/git/gustavoars/linux
git bisect start 'origin/master' 'origin/stable'
# bad: [199d454c0775386a645dd9e80b486c346816762f] Merge remote-tracking branch 'crypto/master'
git bisect bad 199d454c0775386a645dd9e80b486c346816762f
# bad: [0fdcb1ec5bc8e5a40eb6ede7ac1071cfae01a3c0] Merge remote-tracking branch 'hid/for-next'
git bisect bad 0fdcb1ec5bc8e5a40eb6ede7ac1071cfae01a3c0
# bad: [858364b05a8dc9674b93022eab1987b8671c2dc5] Merge remote-tracking branch 'tegra/for-next'
git bisect bad 858364b05a8dc9674b93022eab1987b8671c2dc5
# bad: [69c8ca5334edaa9ec384ab1f474ff2192fde686d] Merge remote-tracking branch 'actions/for-next'
git bisect bad 69c8ca5334edaa9ec384ab1f474ff2192fde686d
# good: [10826e8d36de850875acc8bf1cb8a58b3dd5e313] Merge remote-tracking branch 'dma-mapping/for-next'
git bisect good 10826e8d36de850875acc8bf1cb8a58b3dd5e313
# good: [e7e0da219c21d58170971e07c469a632ee258d94] Merge branches 'arm/soc', 'arm/fixes' and 'arm/drivers' into for-next
git bisect good e7e0da219c21d58170971e07c469a632ee258d94
# good: [98ee6cd3ed6691a2fa8f79f58824604c7bd6d2d3] Merge branches 'for-next/52-bit-kva', 'for-next/cpu-topology', 'for-next/error-injection', 'for-next/psci-cpuidle', 'for-next/rng', 'for-next/smpboot', 'for-next/tbi' and 'for-next/tlbi' into for-next/core
git bisect good 98ee6cd3ed6691a2fa8f79f58824604c7bd6d2d3
# good: [a39e44c197821f1cd54fd191db8454f3d51548ad] Merge remote-tracking branch 'arm/for-next'
git bisect good a39e44c197821f1cd54fd191db8454f3d51548ad
# good: [ff560c0d79d559b1918a4b678eb810e00a578c35] arm64: atomics: Use K constraint when toolchain appears to support it
git bisect good ff560c0d79d559b1918a4b678eb810e00a578c35
# good: [d239a8eb7137458c91ec01c772db5471fd087dd2] Merge branch 'for-next/atomics' into for-next/core
git bisect good d239a8eb7137458c91ec01c772db5471fd087dd2
# good: [fb9c1c1deb5ece57810afdbd743358d02e632d2c] Merge branch 'v4.20/drivers+s900-sps' into next
git bisect good fb9c1c1deb5ece57810afdbd743358d02e632d2c
# good: [236316d6bb486313c49450220ad777036c7a8e6d] Merge branch 'arm/fixes' into for-next
git bisect good 236316d6bb486313c49450220ad777036c7a8e6d
# bad: [afdaf3ac825c02687ec8c74040f3de56f2ad22b1] Merge remote-tracking branch 'arm-soc/for-next'
git bisect bad afdaf3ac825c02687ec8c74040f3de56f2ad22b1
# bad: [529952053dc1b27702fc762abe00f30fe11ce715] Merge remote-tracking branch 'arm64/for-next/core'
git bisect bad 529952053dc1b27702fc762abe00f30fe11ce715
# first bad commit: [529952053dc1b27702fc762abe00f30fe11ce715] Merge remote-tracking branch 'arm64/for-next/core'
```

It also appears to be a KASAN issue, at least according to `tools/testing/ktest/config-bisect.pl`:

```
%%%%%%%% FAILED TO FIND SINGLE BAD CONFIG %%%%%%%%
Hmm, can't make any more changes without making good == bad?
Difference between good (+) and bad (-)
-KASAN_SW_TAGS=n
-KASAN_OUTLINE=y
-CONSTRUCTORS=y
-TEST_KASAN=n
-KASAN_SHADOW_OFFSET=0xdfffffd000000000
-KASAN_INLINE=n
-KASAN_GENERIC=y
 KASAN n -> y
+VMAP_STACK=y
```

I'll try to look into this more over the next few days.

**Comment by nickdesaulniers** [comment]
I think @arndb 's patch resolves this?
https://lore.kernel.org/r/CAKwvOdn90naN2qLx6qBCii67HNOYeJmVqTKEKuUpXcTXLEEaLA@mail.gmail.com/

**Comment by nathanchance** [comment]
Yes, it should. Currently doing a set of builds, will post when the final results come back.

**Comment by nathanchance** [comment]
Arnd sent a v2: https://lore.kernel.org/lkml/20190910115643.391995-1-arnd@arndb.de/

This regressed into mainline because it wasn't picked up, hopefully it can be added before 5.4-rc1.

**Comment by nickdesaulniers** [comment]
fixed: 920fdab7b3ce98c14c840261e364f490f3679a62
